lib/Domain/TextEdit.php

```diff
@@ -36,9 +36,9 @@ public static function applyEdits(array $edits, string $text) : string
         for ($i = \count($edits) - 1; $i >= 0; $i--) {
             $edit = $edits[$i];
 
-            if ($prevEditStart < $edit->start) {
+            if ($prevEditStart < $edit->start || $prevEditStart < $edit->start + $edit->length) {
```

Comment:
You can remove the first condition actually.

Reply:
This is mostly copy/pasted from the MS tolerant parser library, don't think this condition is even tested, but would rather leave as is (in this PR I removed the second condition, but re-added it laster).